internal/cli/cmd/dac/preview/preview.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package preview
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+
+	"github.com/perses/perses/internal/api/utils"
+	persesCMD "github.com/perses/perses/internal/cli/cmd"
+	"github.com/perses/perses/internal/cli/config"
+	"github.com/perses/perses/internal/cli/file"
+	"github.com/perses/perses/internal/cli/opt"
+	"github.com/perses/perses/internal/cli/output"
+	"github.com/perses/perses/internal/cli/resource"
+	"github.com/perses/perses/internal/cli/service"
+	"github.com/perses/perses/pkg/client/api"
+	modelAPI "github.com/perses/perses/pkg/model/api"
+	modelV1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/prometheus/common/model"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type previewResponse struct {
+	Project   string `json:"project" yaml:"project"`
+	Dashboard string `json:"dashboard" yaml:"dashboard"`
+	Preview   string `json:"preview" yaml:"preview"`
+}
+
+func newEphemeralDashboard(project string, name string, ttl model.Duration, dashboard *modelV1.Dashboard) *modelV1.EphemeralDashboard {
+	return &modelV1.EphemeralDashboard{
+		Kind: modelV1.KindEphemeralDashboard,
+		Metadata: modelV1.ProjectMetadata{
+			Metadata:               modelV1.Metadata{Name: name},
+			ProjectMetadataWrapper: modelV1.ProjectMetadataWrapper{Project: project},
+		},
+		Spec: modelV1.EphemeralDashboardSpec{
+			EphemeralDashboardSpecBase: modelV1.EphemeralDashboardSpecBase{TTL: ttl},
+			DashboardSpec:              dashboard.Spec,
+		},
+	}
+}
+
+type option struct {
+	persesCMD.Option
+	opt.ProjectOption
+	opt.FileOption
+	opt.DirectoryOption
+	opt.OutputOption
+	dashboards        []*modelV1.Dashboard
+	writer            io.Writer
+	apiClient         api.ClientInterface
+	ttl               model.Duration
+	ttlAsAString      string
+	pullRequestNumber string
```

Comment:
I'm wondering if we shouldnt go for a single and generic "prefix" flag instead, so people put whatever they want (PR number & repo name looks like standard use cases, but maybe for local work they want to put their username or whatever)

Reply:
yeah I agree. I have replaced these flags with the prefix flag